aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -116,6 +116,9 @@ Tensor empty_cpu(IntArrayRef size, const TensorOptions& options) {
   if (size.size() != 1 || size[0] != 0) {
     tensor.unsafeGetTensorImpl()->set_sizes_contiguous(size);
   }
+
+  auto memory_format = optional_memory_format.value_or(MemoryFormat::Contiguous);
```

Comment:
Something I find perplexing here, is that if the optional memory format defaults to `Contiguous`, why not just have a non-optional memory format with default value `Contiguous`?

Reply:
The answer, I guess, is because you use this to test for error condition in `empty_out`?

Reply:
Is there some downstream codegen requirement that the out and non-out overloads have to exactly match?

Reply:
Exactly there is codegen requirement that the out and non-out overloads have to exactly match. 
Having optional is required for 'out' cases and sparse tensors as well, when any of MemoryFormat values is not valid.

Reply:
Can we get a comment here to that effect?

Reply:
Probably just something like, "Preferably, this argument would not be accepted by `_out`, but the code generator requires the out and non-out overloads to match exactly"